client/src/App.js

```diff
@@ -0,0 +1,116 @@
+    // this.state = {
+    //   name: "Ammar"
+    // }
+    this.state = {};
+    this.getData = this.getData.bind(this);
+    this.sendData = this.sendData.bind(this);
+    this.updateData = this.updateData.bind(this);
+    this.deleteData = this.deleteData.bind(this);
+  }
+
+  componentDidMount() {
+
+  }
+
+  getData() {
+    axios.get('/api/times')
+      .then(response => {
+        let data = response.data;
+        this.setState({ data });
+        console.log(this.state.data);
+      });
+  }
+
+  sendData(userData) {
+    axios.post('/api/times', userData)
+      .then(response => {
+        console.log(response);
+      })
+  }
+
+  updateData(userData) {
+    axios.put('/api/times?id=' + userData['id'] + '&name=' + userData['name'])
+      .then(response => {
+        console.log(response);
+      })
+  }
+
+  deleteData(userData) {
+    axios.delete('/api/times/' + userData)
+      .then(response => {
+        console.log(response);
+      })
+  }
+
+  render() {
+    return (
+      <div className="App">
+        <div className="App-header">
+          <img src={logo} className="App-logo" alt="logo" />
+          {/*<h2>{welcomeMsg()}</h2>*/}
+          <WelcomeUser name="Ammar" />
+          {/*<h2>Hey {this.state.data.name}, Welcome to REact learning</h2>*/}
+
+          <button type="btn" className="btn btn-primary" onClick={this.getData}>Fetch Data</button>
+          {/*<TableData />*/}
+          <div className="medium-space"></div>
+          <div className="clearfix"></div>
+          <button type="btn" className="btn btn-primary" onClick={() => this.sendData({ "name": "Dan", "role": "Manager" })}>Send Data</button>
+          <div className="medium-space"></div>
+          <div className="clearfix"></div>
+          <button type="btn" className="btn btn-primary" onClick={() => this.updateData({ "id": 4, "name": "John" })}>Update Data</button>
+          <div className="medium-space"></div>
+          <div className="clearfix"></div>
+          <button type="btn" className="btn btn-primary" onClick={() => this.deleteData('4')}>Delete Data</button>
+
+        </div>
+      </div>
+    );
+  }
+}
+
+// function welcomeMsg() {
+//   return "Welcome to REact"
+// }
+
+function WelcomeUser(props) {
+  return <h2>Hey {props.name}, Welcome to REact learning</h2>;
+}
+
+class TableData extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+    }
+  }
+
+  componentDidMount() {
+    axios.get('/api/times')
+      .then(response => {
+        let data = response.data;
+        this.setState({ data });
+        console.log(this.state.data);
+      });
+  }
+
+  render() {
+    return (
+      <BootstrapTable data={this.state.data}>
+        <TableHeaderColumn dataField='id' isKey={true}>Emp Id</TableHeaderColumn>
+        <TableHeaderColumn dataField='name'>Employee Name</TableHeaderColumn>
+        <TableHeaderColumn dataField='role'>Employee Role</TableHeaderColumn>
+      </BootstrapTable>
+    )
```

Comment:
I know this part isn't finished yet, we can discuss this over hangout. 
there is a nice technique in react since you can use any kind of javascript within { brackets }:
```jsx
const userRow = user => (
  <tr>
    <td>{user.name}</td>
    <td>{new Date(user.time).toLocaleTimeString()}</td>
    <td>{user.id}</td>
  </tr>
)

 return (
 <table>
  <thead>
    <tr><th>... header cols goes here</th></tr>
  </thead>
  <tbody>
    {users.map(userRow)}
  </tbody>
  </table>
);
```